docs/website/blog/2022-06-30-binding-database-service-without-sbo.md

```diff
@@ -0,0 +1,301 @@
+---
+title: Binding to a database service without the Service Binding Operator
+author: Parthvi Vala
+author_url: https://github.com/valaparthvi
+author_image_url: https://github.com/valaparthvi.png
+tags: ["binding"]
+slug: binding-database-service-without-sbo
+---
+
+How to bind your application to a database service without the Service Binding Operator.
+
+<!--truncate-->
+
+There are a few ways of binding your application to a database service with the help of odo. The recommended way is with the help of Service Binding Operator(SBO), but it is also possible to bind without it, and this blog will show you how.
+
+
+## Architecture
+We have a simple CRUD application built in Go that can create/list/update/delete a place. This application requires connecting to a MongoDB database in order to function correctly, which will be deployed as a microservice on the cluster.
+
+## Prerequisites:
+This blog assumes:
+- [odo v3.0.0-beta1](https://github.com/redhat-developer/odo/releases/tag/v3.0.0-beta1)
+- you have access to a Kubernetes or OpenShift cluster to be able to install the MongoDB operator.
+- you have _Helm_ installed on your system. See https://helm.sh/docs/intro/install/ for installation instructions.
+
+## (Optional) Setting up the namespace
+0. We will create a new namespace to deploy our application in, with the help of odo.
+```sh
+odo create namespace restapi-mongodb
+```
+
+## Setting up the MongoDB microservice
+We are going to use the Bitnami's helm charts for creating our MongoDB database.
+
+1. Add the Bitnami's Helm charts repository and make your Helm client up to date with it:
+```sh
+helm repo add bitnami https://charts.bitnami.com/bitnami && helm repo update
+```
+
+2. Export the necessary environment variables:
+```sh
+export MY_MONGODB_ROOT_USERNAME=root
```

Comment:
I'm pretty sure the exports are not necessary.

They would be necessary if you used `odo dev --var MY_MONGODB_ROOT_USERNAME`, but using `odo dev --var PASSWORD=$MY_MONGODB_ROOT_USERNAME `, you just need to declare the variables into the shell session, without exporting them.

Note that exporting a variable (especially a password) into a shell gives access to it from *all* the processes you start from your shell. This can be a security breach, it must be used cautiously.